src/components/magnet/vec-utils.ts

```diff
@@ -11,9 +11,11 @@ export class Vector {
     return new Vector(-this.x, -this.y);
   }
 
-  public add(v: Vector | number) {
+  public add(v: Vector | number, y?: number) {
```

Comment:
:+1: ok fixing it.

I was a little worried in general about needing to use so many `new Vector` calls in the middle of an inner-loop calculation that gets called sometimes thousands of times, but it seems like it's mostly ok. If we wanted to improve performance, though, doing this entirely on an array would probably be faster, at the expense of a bit of legibility. But the calculation doesn't seem like the biggest bottleneck (rendering the lines seems slower) so it might not make a difference.